mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraUidProvider.java

```diff
@@ -66,21 +66,20 @@ public CassandraUidProvider(Session session) {
     }
 
     @Override
-    public long nextUid(MailboxSession mailboxSession, Mailbox mailbox) throws MailboxException {
+    public MessageUid nextUid(MailboxSession mailboxSession, Mailbox mailbox) throws MailboxException {
         CassandraId mailboxId = (CassandraId) mailbox.getMailboxId();
-        if (findHighestUid(mailboxId).isFirst()) {
-            Optional<Uid> optional = tryInsertUid(mailboxId, FIRST_UID);
+        if (! findHighestUid(mailboxId).isPresent()) {
+            Optional<MessageUid> optional = tryInsertUid(mailboxId, com.google.common.base.Optional.absent());
```

Comment:
Why the use of Guava optionals here ? Isn't the use of Optional internal to the class ? I don't see the value added by the use of Guava optional....